tests/integration/test_charm.py

```diff
@@ -65,3 +65,29 @@ async def test_ingress_reachable(app: Application):
     response = requests.get("http://hello-kubecon")
 
     assert response.status_code == 200
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_owasp_modsecurity_crs(app: Application):
+    """
+    arrange: given charm has been built, deployed, related to a dependent application
+    and owasp-modsecurity-crs is set to True
+    act: when the dependent application is queried via the ingress with malicious request
+    assert: then the response is HTTP 403 Forbidden.
+    """
+    # Get the IP address which is in the status message
+    ip_regex = r"[0-9]+(?:\.[0-9]+){3}"
+    ip_address_match = re.findall(ip_regex, app.units[0].workload_status_message)
+    assert (
+        ip_address_match
+    ), f"could not find IP address in status message: {app.units[0].workload_status_message}"
+    ip_address = ip_address_match[0]
+    ps = subprocess.Popen(["echo", f"{ip_address} hello-kubecon"], stdout=subprocess.PIPE)  # nosec
+    subprocess.run(["sudo", "tee", "-a", "/etc/hosts"], stdin=ps.stdout)  # nosec
```

Comment:
Out of curiosity, does the `stdin` have to be the `stdout` from calling another process? Or could it just be the string `f"{ip_address} hello-kubecon"`?

Reply:
I'll do a test with the "input" parameter.
https://docs.python.org/3/library/subprocess.html#subprocess.run

Reply:
That does look better 👍